consensus/src/chained_bft/network.rs

```diff
@@ -320,20 +267,47 @@ impl ConsensusNetworkImpl {
     }
 }
 
-struct NetworkTask<T, S> {
+pub struct NetworkTask<T> {
     proposal_tx: channel::Sender<ProposalMsg<T>>,
     vote_tx: channel::Sender<VoteMsg>,
     block_request_tx: channel::Sender<BlockRetrievalRequest<T>>,
     sync_info_tx: channel::Sender<(SyncInfo, AccountAddress)>,
-    all_events: S,
-    epoch_mgr: Arc<EpochManager>,
+    all_events: Box<dyn Stream<Item = failure::Result<Event<ConsensusMsg>>> + Send + Unpin>,
+    validators: Arc<ValidatorVerifier>,
```

Comment:
I'd like to but it's hard to create a new task given we need to pass the `all_events` to the new task since it can't be Cloned.

So my tentative plan is to replace the validators, but we'll see.